core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -735,9 +735,9 @@ private[log] class Cleaner(val id: Int,
        *   1) if there exists a message with the same key but higher offset
        *   2) if the message is a delete "tombstone" marker and enough time has passed
        */
-      val redundant = foundOffset >= 0 && record.offset < foundOffset
+      val latestOffsetForKey = record.offset() >= foundOffset
       val obsoleteDelete = !retainDeletes && !record.hasValue
```

Comment:
I find this a bit hard to follow also. Is it any clearer if we invert it as well?
```scala
val latestOffsetForKey = record.offset() >= foundOffset
val isRetainedValue = record.hasValue || retainDeletes
latestOffsetForKey && isRetainedValue 
```
Unfortunately I couldn't think of a great name for it.

Reply:
This change makes sense to me because I think logical not operator is sometimes difficult to follow.

Reply:
I push a new commit to invert obsoleteDelete  and change comments to make them conform to the code